drivers/volume/portworx/portworx.go

```diff
@@ -2890,6 +2931,37 @@ func getSDKContainerPort() (int32, error) {
 	return 0, fmt.Errorf("px-sdk target port not found in service")
 }
 
+func getImageList(endpointURL, pxVersion, k8sVersion string) (map[string]string, error) {
+	var imageList map[string]string
+	client := &http.Client{}
+	u, err := url.Parse(endpointURL)
+	if err != nil {
+		return imageList, fmt.Errorf("failed to parse URL %s request. Cause: %v", endpointURL, err)
+	}
+	q := u.Query()
+	q.Set("kbver", k8sVersion)
+	u.RawQuery = q.Encode()
+	u.Path = path.Join(pxVersion, "version")
+	resp, err := client.Get(u.String())
+	if err != nil {
+		return imageList, fmt.Errorf("error while downloading version file from %s. Cause: %v", u.String(), err)
+	}
+	defer resp.Body.Close()
+	body, err := io.ReadAll(resp.Body)
+
+	yamlMap := make(map[string]interface{})
+	if err := yaml.Unmarshal(body, &yamlMap); err != nil {
+		return imageList, err
+	}
+
+	imageList = make(map[string]string)
+	for key, value := range yamlMap["components"].(map[interface{}]interface{}) {
+		imageList[key.(string)] = value.(string)
+	}
+	imageList["version"] = fmt.Sprintf("portworx/oci-monitor:%s", yamlMap["version"].(string))
+	return imageList, nil
```

Comment:
I think it would be good to do a debug print of what the test actually gets from the versions URL, it will help us debug later incase of errors